examples/util/WebNNRunner.js

```diff
@@ -0,0 +1,547 @@
+    // Override by inherited if needed
+    const typedArray = this._currentModelInfo.isQuantized || false ? Uint8Array : Float32Array;
+    return typedArray;
+  };
+
+  /**
+   * This method is to init set '_inputTensor'.
+   */
+  _initInputTensor = () => {
+    const typedArray = this._getInputTensorTypedArray();
+    this._inputTensor = [new typedArray(this._currentModelInfo.inputSize.reduce((a, b) => a * b))];
+  };
+
+  /**
+   * This method is to typedArray type for outputTensor.
+   * @returns {function} This returns Uint8Array or Float32Array function or other typedArray function if inherited.
+   */
+  _getOutputTensorTypedArray = () => {
+    // Override by inherited if needed
+    const typedArray = this._currentModelInfo.isQuantized || false ? Uint8Array : Float32Array;
+    return typedArray;
+  };
+
+  /**
+   * This method is to init set '_outputTensor'.
+   */
+  _initOutputTensor = () => {
+    // Override by inherited if needed
+    const typedArray = this._getOutputTensorTypedArray();
+    const outputSize = this._currentModelInfo.outputSize;
+
+    if (typeof outputSize === 'number') {
+      this._outputTensor = [new typedArray(outputSize)];
+    } else {
+      this._outputTensor = [new typedArray(outputSize.reduce((a, b) => a * b))];
+    }
+  };
+
+  _doInitialization = (modelInfo) => {
+    this._setLoadedFlag(false);
+    this._setInitializedFlag(false);
+    this._setBackend(null);
+    this._setPrefer(null);
+    this._setModelRequiredOps(new Set());
+    this._setDeQuantizeParams([]);
+    this._setSubgraphsSummary([]);
+    this._setModelInfo(modelInfo);
+    this._initInputTensor();
+    this._initOutputTensor();
+  };
+
+  _checkInitializedCompilation = (options) => {
+    return this._bInitialized && this._currentBackend === options.backend && this._currentPrefer === options.prefer;
+  }
+
+  _doCompile = async (options) => {
+    let model = null;
+    const backend = options.backend;
+    const prefer = options.prefer;
+
+    this._setBackend(backend);
+    this._setPrefer(prefer);
+
+    const postOptions = this._currentModelInfo.postOptions || {};
+    const configs = {
+      rawModel: this._rawModel,
+      backend: this._currentBackend,
+      prefer: this._currentPrefer,
+      softmax: postOptions.softmax || false,
+      inputScaleFactor: options.scaleFactor, // for GNA
+    };
+
+    switch (this._rawModel._rawFormat) {
+      case 'TFLITE':
+        model = new TFliteModelImporter(configs);
+        break;
+      case 'ONNX':
+        model = new OnnxModelImporter(configs);
+        break;
+      case 'OPENVINO':
+        model = new OpenVINOModelImporter(configs);
+        break;
+      default:
+        throw new Error(`Unsupported '${rawModel._rawFormat}' model.`);
+    }
+
+    this._setModel(model);
+    this._model.setSupportedOps(this._supportedOps);
+    this._model.setEagerMode(this._bEagerMode);
+    await this._model.createCompiledModel();
+  };
+
+  _saveDetails = () => {
```

Comment:
Can it be called in `_doCompile` of this class? With that, we don't need this override method defined in `BaseRunner`.